src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -66,18 +66,18 @@ ss::future<client_ptr> kafka_client_cache::fetch_or_insert(
                 auto item = inner_list.back();
                 vlog(plog.debug, "Cache size reached, evicting {}", item.key);
                 inner_list.pop_back();
-                _evicted_items.push_back(std::move(item));
-
-                // If the timer is not armed, then trigger it for a few seconds
-                // from now. If the timer is armed and it won't run until far
-                // into the future, then trigger it a few seconds from now. If
-                // the timer is armed and it will run soon, then do nothing.
-                auto window = ss::lowres_clock::now() + 1s;
-                if (
-                  !_clean_and_evict_timer.armed()
-                  || (_clean_and_evict_timer.armed() && _clean_and_evict_timer.get_timeout() > window)) {
-                    _clean_and_evict_timer.rearm(window);
-                }
+
+                co_await ss::try_with_gate(_gate, [item{std::move(item)}] {
+                    return item.client->stop().handle_exception(
+                      [k{item.key}](std::exception_ptr ex) {
+                          // The stop failed
+                          vlog(
+                            plog.debug,
+                            "Client {} stop already happened {}",
+                            k,
+                            ex);
+                      });
+                });
```

Comment:
I still need to figure out where to put the lock. For some reason my brain is having trouble conceptualizing this.